### PR TITLE
Introduction of COSM WASM smartcontracts to chain

### PR DESCRIPTION
**Pull request by ntindle**
## Changes

* Addition of Comwasm
* Go Mod Upgraded to verison go 1.17 for wasm dependencies
* Addition of Custom Antehandler With SignatureVerification Disabled
* Addition of `testing_wasm` to add a setup for testing Cosmwasm Integration On Sonr Node.

## API Updates

* Addition Of New wasm based Query

### New Features *(required)*

* Addition Of Cosm-wasm Support

### Deprecations *(required)*

* None

### Enhancements *(optional)*

* Addition Of Cosm-wasm Support

## Checklist

* [ ] Unit tests
* [ ] Documentation

## References *(optional)*

Fixes
Connects

 Mention [stepsize] in a comment if you'd like to report some technical debt. See examples here.

### CODE REVIEW

**Comment by ntindle** [comment]
@utkush check to make sure that your whitespace changes aren't causing changes in every single file

File: testing_wasm/package.json [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "testing",
+  "version": "1.0.0",
+  "description": "Testing",
+  "scripts": {
+    "start": "node -r dotenv/config index.js",
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "Utkarsh Varma",
+  "license": "ISC",
```
Comment (ntindle) [inline testing_wasm/package.json]: NO LICENSE

Comment (utkush) [reply testing_wasm/package.json]: fixed.

Comment (ntindle) [reply testing_wasm/package.json]: Sorry to be clear should be "license": "NO LICENSE"

File: app/app.go [hunk 2]
```diff
@@ -113,16 +119,60 @@ import (
...
+// produce a list of enabled proposals to pass into wasmd app.
+func GetEnabledProposals() []wasm.ProposalType {
+	if EnableSpecificProposals == "" {
+		if ProposalsEnabled == "true" {
+			return wasm.EnableAllProposals
+		}
+		return wasm.DisableAllProposals
+	}
+	chunks := strings.Split(EnableSpecificProposals, ",")
+	proposals, err := wasm.ConvertToProposals(chunks)
```
Comment (joshLong145) [inline app/app.go:154]: can this ever throw an error on the parse?

File: testing_wasm/package.json [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  "dependencies": {
+    "@cosmjs/stargate": "^0.28.4",
+    "@cosmjs/cosmwasm-stargate": "^0.28.4",
+    "@cosmjs/proto-signing": "^0.28.4",
+    "chai": "^4.3.6",
+    "node-fetch": "^3.2.5",
+    "protobufjs": "^6.10.2"
+  },
+  "devDependencies": {
+    "@terra-money/terra.js": "^2.0.16",
```
Comment (joshLong145) [inline testing_wasm/package.json]: why do we need this "terra.js" dep?

**Review by joshLong145 — APPROVED** [review]
left some comments nice work

File: testing_wasm/package.json (hunk 3 above)
Comment (UvRoxx) [reply testing_wasm/package.json]: Sorry the testing setup is from an old config I will remove it

File: testing_wasm/package.json (hunk 1 above)
Comment (UvRoxx) [reply testing_wasm/package.json]: Fixed.

**Comment by ntindle** [comment]
looks like its failing build https://github.com/sonr-io/sonr/actions/runs/3107748973/jobs/5036198211

File: go.mod [hunk 4]
```diff
@@ -1,9 +1,9 @@
 module github.com/sonr-io/sonr
 
-go 1.16
+go 1.17
 
 require (
-	github.com/99designs/keyring v1.1.6
```
Comment (mcjcloud) [inline go.mod:6]: This dep should not have been removed. I ran `go mod tidy` locally with your current changes and it automatically added it back, can you try this?

File: pkg/motor/create_account.go [hunk 5]
```diff
@@ -126,6 +127,7 @@ func (mtr *motorNodeImpl) CreateAccountWithKeys(request mt.CreateAccountWithKeys
 		pskShard,
 		pwShard,
 	)
+	fmt.Println("Response From Create Vault :", vaultService)
```
Comment (mcjcloud) [inline pkg/motor/create_account.go:130]: Was this just for debugging?

File: pkg/vault/vault.go [hunk 6]
```diff
@@ -17,7 +17,7 @@ type vaultImpl struct {
 
 func New() VaultClient {
 	return &vaultImpl{
-		vaultEndpoint: "https://vault.sonr.ws",
-		// vaultEndpoint: "http://127.0.0.1:1234",
+		//vaultEndpoint: "https://vault.sonr.ws",
+		vaultEndpoint: "http://127.0.0.1:1234",
```
Comment (mcjcloud) [inline pkg/vault/vault.go]: change this back

**Review by mcjcloud — CHANGES_REQUESTED** [review]

**Comment by UvRoxx** [comment]
Working of fixing the compile issues for target  linux:amd64

File: pkg/motor/create_account.go (hunk 5 above)
Comment (UvRoxx) [reply pkg/motor/create_account.go:130]: Yes will remove it and cleanup

File: go.mod (hunk 4 above)
Comment (UvRoxx) [reply go.mod:6]: fixed

**Comment by ntindle** [comment]
> Working of fixing the compile issues for target linux:amd64

I think its happening across all targets for building and testing. The other half of the runners test the motor methods

File: .github/workflows/build-and-test.yml [hunk 7]
```diff
@@ -46,6 +46,45 @@ jobs:
...
+      - name: Run tests and evaluate coverage
+        continue-on-error: true
+        run: go test -race -coverprofile=coverage.out -covermode=atomic ./...
+
+      - name: Run simulate
+        continue-on-error: true
+        run: go test -benchmem -run=^$ -bench ^BenchmarkSimulation ./app -NumBlocks=200 -BlockSize 50 -Commit=true -Verbose=true -Enabled=true
+
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v3
```
Comment (ntindle) [inline .github/workflows/build-and-test.yml]: Revert the changes to this file, theres better ways to do this

File: .github/workflows/build-and-test.yml [hunk 8]
```diff
@@ -30,10 +30,10 @@ jobs:
         run: |
           sudo curl https://get.ignite.com/cli@${{matrix.ignite-version}}! | sudo bash
 
-      - name: Run build against Release target for macos and linux
+      - name: Run build against Release target for linux
         continue-on-error: false
         run: |
-          ignite chain build --release -t linux:amd64 -t darwin:amd64
+          ignite chain build --release -t linux:amd64
```
Comment (ntindle) [inline .github/workflows/build-and-test.yml]: I would duplicate this step and add an if statement

something like 

```yaml
      - name: Run build against Release target for macos
        if: maxtrix.os == macos-latest
        continue-on-error: false
        run: |
          ignite chain build --release -t darwin:amd64 darwin:arm64
      - name: Run build against Release target for ubuntu
        if: maxtrix.os == ubuntu-latest
        continue-on-error: false
        run: |
          ignite chain build --release -t linux:amd64 linux:arm64
```

File: pkg/vault/vault.go (hunk 6 above)
Comment (UvRoxx) [reply pkg/vault/vault.go]: fixed

File: app/app.go (hunk 2 above)
Comment (UvRoxx) [reply app/app.go:154]: The const if defined incorrectly would throw such an error but if this error exists, it would always fail to build so it won't get to production.

**Review by joshLong145 — APPROVED** [review]

File: .github/workflows/build-and-test.yml [hunk 9]
```diff
@@ -30,10 +30,17 @@ jobs:
...
-      - name: Run build against Release target for macos and linux
+      - name: Run build against Release target for macos
+        if: ${{matrix.os == 'macos-latest'}}
         continue-on-error: false
         run: |
-          ignite chain build --release -t linux:amd64 -t darwin:amd64
+          ignite chain build --release -t darwin:amd64
+
+      - name: Run build against Release target for ubuntu
+        if: ${{matrix.os == 'ubuntu-latest'}}
```
Comment (ntindle) [inline .github/workflows/build-and-test.yml:40]: Add this line to the Upload coverage to Codecov task too. Don't want to waste time

**Comment by ntindle** [comment]
@mcjcloud plz re-review

File: app/ante.go [hunk 10]
```diff
@@ -0,0 +1,67 @@
+package app
```
Comment (prdlk) [inline app/ante.go:1]: Mad hype custom ante handler

**Review by prdlk — APPROVED** [review]
